controllers/machine_controller_phases.go

```diff
@@ -197,6 +197,14 @@ func (r *MachineReconciler) reconcileInfrastructure(ctx context.Context, m *clus
 	if infraConfig == nil && err == nil {
 		return nil
 	} else if err != nil {
+		if m.Status.InfrastructureReady {
+			// Infra or bootstrap object went missing after the machine was up and running
+			r.Log.Error(err, "Infrastructure or bootstrap object went missing. Updating Machine Status")
+			machineStatusError := capierrors.MachineStatusError("InvalidConfiguration")
+			m.Status.ErrorReason = &machineStatusError
+			m.Status.ErrorMessage = pointer.StringPtr(fmt.Sprintf("Failure detected from referenced resource %v with name %q",
```

Comment:
What kind of failure? Let's add a little more color on why we're setting this value